### PR TITLE
enable selecting test folder when using nbresult in notebooks

### PR DESCRIPTION
**Pull request by brunolajoie**
The following PR updates nbresult (without breaking it)

## Before
Previously, we could only use nbresult in notebooks at the **root**
image

## After
Now, we can specify in which `subdir` folder to store the `pickle` and where to read the `pickle` and the `test_xxx.py`
image

Demo repo as per screenshot [here](https://github.com/brunolajoie/nbresultdemo/) 

## Why need this ?
This can be useful for taxifare-model project, where we regroup all tests of the week in the `./tests` with 1 subdir per challenge, some test launched from notebooks in `./notebooks`, while some other are launched from `Makefile` and targetting python code located in `./taxifare_model/ml_logic` for instance

### CODE REVIEW

**Comment by brunolajoie** [comment]
- What do you think about the unique Makefile command problem @gmanchon ?
- What do you think about the idea @krokrob ? (before diving into the code)

**Comment by krokrob** [comment]
What does the `make` command returns (before diving into the code) ? cc @brunolajoie

**Comment by brunolajoie** [comment]
Edit: actually, Make works!  Maybe I was too tired yesterday :) 
image

**Comment by brunolajoie** [comment]
Actually, Make works as well! I was maybe too tired yesterday when I tried
image

File: nbresult/__init__.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import pickle
 import os
+from os.path import dirname, join, abspath
```
Comment (krokrob) [inline nbresult/__init__.py:3]: To be removed

File: nbresult/__init__.py [hunk 2]
```diff
@@ -12,41 +15,88 @@ class ChallengeResult:
...
+        while not os.path.isdir(join(cwd, 'tests')):
+            cwd = dirname(cwd)
+            if cwd == os.sep:
+                raise NameError(
+                    "Could not find tests dir in any parent folder")
+
+        tests_path = join(cwd, 'tests')
+
+        if self.subdir is not None:
+            tests_path = join(tests_path, self.subdir)
```
Comment (krokrob) [inline nbresult/__init__.py:37]:
```suggestion
        while not os.path.isdir(os.path.join(cwd, 'tests')):
            cwd = os.path.dirname(cwd)
            if cwd == os.sep:
                raise NameError(
                    "Could not find /tests directory in any parent folder")

        tests_path = os.path.join(cwd, 'tests')

        if self.subdir is not None:
            tests_path = os.path.join(tests_path, self.subdir)
```

File: nbresult/__init__.py [hunk 3]
```diff
@@ -12,41 +15,88 @@ class ChallengeResult:
...
+
     def write(self):
         """Write down values from initialize to result.pickle"""
+        tests_path = self._locate_tests()
+
         if sys.getsizeof(self) > 10_000:
             raise ValueError(f"""Check the arguments of your ChallengeResult
                 {self.name}, one is way too big.""")
-        result_file = os.path.join(os.getcwd(), "tests", f"{self.name}.pickle")
+        result_file = join(tests_path, f"{self.name}.pickle")
```
Comment (krokrob) [inline nbresult/__init__.py:48]:
```suggestion
        result_file = os.path.join(tests_path, f"{self.name}.pickle")
```

File: nbresult/__init__.py [hunk 4]
```diff
@@ -12,41 +15,88 @@ class ChallengeResult:
...
+        cwd = os.getcwd()
+
+        while not os.path.isdir(join(cwd, 'tests')):
+            cwd = dirname(cwd)
+            if cwd == os.sep:
+                raise NameError(
+                    "Could not find tests dir in any parent folder")
+
+        pickle_path = glob.glob(
+            abspath(join(cwd, '**', f'{name}.pickle')), recursive=True)[0]
```
Comment (krokrob) [inline nbresult/__init__.py:93]:
```suggestion
        while not os.path.isdir(os.path.join(cwd, 'tests')):
            cwd = os.path.dirname(cwd)
            if cwd == os.sep:
                raise NameError(
                    "Could not find /tests directory in any parent folder")

        pickle_path = glob.glob(
            os.path.abspath(os.path.join(cwd, '**', f'{name}.pickle')), recursive=True)[0]
```

**Review by krokrob — CHANGES_REQUESTED** [review]
Great improvement @brunolajoie , I have some suggestion

File: nbresult/__init__.py (hunk 1 above)
Comment (brunolajoie) [inline nbresult/__init__.py:3]:
```suggestion
```

**Comment by brunolajoie** [comment]
- comments fixed ✅ 
- tested again on nnresultdemo ✅ 

kev feel free to merge

**Comment by brunolajoie** [comment]
@gmanchon let's merge?

**Review by krokrob — APPROVED** [review]

**Comment by krokrob** [comment]
AWESOME @brunolajoie

**Pull request merged**